troi/patches/periodic_jams.py

```diff
@@ -45,26 +53,47 @@ def outputs():
 
     @staticmethod
     def slug():
-        return "daily-jams"
+        return "periodic-jams"
 
     @staticmethod
     def description():
-        return "Generate a daily playlist from the ListenBrainz recommended recordings."
+        return "Generate a periodic playlist from the ListenBrainz recommended recordings."
 
     def create(self, inputs):
         user_name = inputs['user_name']
         jam_date = inputs.get('jam_date')
         if jam_date is None:
             jam_date = datetime.utcnow().strftime("%Y-%m-%d %a")
+        jam_type = inputs.get('type')
+        if jam_type is None:
+            jam_type = self.JAM_TYPES[0]
+        else:
+            jam_types = jam_type.lower()
+            if jam_type not in self.JAM_TYPES:
+                raise RuntimeError("Jam type must be one of %s" % ", ".join(jam_types))
 
         recs = troi.listenbrainz.recs.UserRecordingRecommendationsElement(user_name, "raw", count=1000, auth_token=inputs.get("token"))
 
         recent_listens_lookup = troi.listenbrainz.listens.RecentListensTimestampLookup(user_name, days=2, auth_token=inputs.get("token"))
         recent_listens_lookup.set_sources(recs)
 
-        # Remove tracks that have not been listened to before.
-        never_listened = troi.filters.NeverListenedFilterElement()
-        never_listened.set_sources(recent_listens_lookup)
+        if jam_type in ("daily-jams", "weekly-jams"):
+            # Remove tracks that have not been listened to before.
+            never_listened = troi.filters.NeverListenedFilterElement()
+            never_listened.set_sources(recent_listens_lookup)
+            if jam_type == "daily-jams":
+                jam_name = "Daily Jams"
+            else:
+                jam_name = "Weekly Jams"
+                jam_date = "week of " + jam_date
+        elif jam_type == "weekly-new-jams":
+            # Remove tracks that have been listened to before.
+            never_listened = troi.filters.NeverListenedFilterElement(remove_unlistened=False)
```

Comment:
The NeverListenedFilterElement could probably be renamed to something else and the name of the parameter as well but I can't think of a good alternative so fine to do later I guess.